lnd_test.go

```diff
@@ -246,6 +284,82 @@ func testBasicChannelFunding(net *networkHarness, t *harnessTest) {
 	closeChannelAndAssert(t, net, ctxt, net.Alice, chanPoint, false)
 }
 
+// testFundingPersistence is intended to ensure that the Funding Manager
+// persists the state of new channels prior to broadcasting the channel's
+// funding transaction. This ensures that the daemon maintains an up-to-date
+// representation of channels if the system is restarted or disconnected.
+// testFundingPersistence mirrors testBasicChannelFunding, but adds restarts
+// and checks for the state of channels with unconfirmed funding transactions.
+func testChannelFundingPersistence(net *networkHarness, t *harnessTest) {
+	timeout := time.Duration(time.Second * 15)
+	ctxb := context.Background()
+	ctxt, _ := context.WithTimeout(ctxb, timeout)
+
+	chanAmt := btcutil.Amount(btcutil.SatoshiPerBitcoin / 2)
+	pushAmt := btcutil.Amount(0)
+
+	// Create a new channel, then broadcast the funding transaction.
+	pendingUpdate, err := net.OpenPendingChannel(ctxt, net.Alice, net.Bob, chanAmt, pushAmt, 1)
+	if err != nil {
+		t.Fatalf("unable to open channel: %v", err)
+	}
+
+	// At this point, the channel's funding transaction will have
+	// been broadcast, but not confirmed. Alice and Bob's nodes
+	// should reflect this when queried via RPC.
+	assertNumChannelsPending(t, ctxt, net.Alice, net.Bob, 1)
+
+	// Restart both nodes to test that they both recover gracefully.
+	// TODO(bvu): add restart for Bob.
```

Comment:
I took out the restart for Bob because if I shutdown and restart both Alice and Bob simultaneously, there seems to be some flakiness with the reconnect. I get an "unable to find peer" during the following test. Any ideas on how to mitigate that?